src/Neo.CLI/CLI/MainService.CommandLine.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2016-2023 The Neo Project.
+//
+// The neo-cli is free software distributed under the MIT software
+// license, see the accompanying file LICENSE in the main directory of
+// the project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.NamingConventionBinder;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+
+namespace Neo.CLI
+{
+    public partial class MainService
+    {
+        public int OnStartWithCommandLine(string[] args)
+        {
+            RootCommand rootCommand = new(Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyTitleAttribute>()!.Title)
+            {
+                new Option<string>(new[] { "-c", "--config","/config" }, "Specifies the config file."),
+                new Option<string>(new[] { "-n", "--network","/network" }, "Indicates the network of the chain [mainnet, testnet, privnet]."),
+                new Option<string>(new[] { "-w", "--wallet","/wallet" }, "The path of the neo3 wallet [*.json]."),
+                new Option<string>(new[] { "-p", "--password" ,"/password" }, "Password to decrypt the wallet, either from the command line or config file."),
+                new Option<string>(new[] { "--db-engine","/db-engine" }, "Specify the db engine."),
+                new Option<string>(new[] { "--db-path","/db-path" }, "Specify the db path."),
+                new Option<string>(new[] { "--noverify","/noverify" }, "Indicates whether the blocks need to be verified when importing."),
+                new Option<string[]>(new[] { "--plugins","/plugins" }, "The path of the plugin [plugin1 plugin2]."),
+            };
+
+            rootCommand.Handler = CommandHandler.Create<RootCommand, CommandLineOptions, InvocationContext>(Handle);
+            return rootCommand.Invoke(args);
+        }
+
+        private void Handle(RootCommand command, CommandLineOptions options, InvocationContext context)
+        {
+            Start(options);
+        }
+
+        private static void CustomProtocolSettings(CommandLineOptions options, ProtocolSettings settings)
+        {
+            uint network = settings.Network;
+            ProtocolSettings tempSetting = settings;
+            // if specified network without specifying config, then load the default config
+            if (!string.IsNullOrEmpty(options.Network))
+            {
+                if (!uint.TryParse(options.Network, out network))
+                {
+                    network = options.Network switch
```

Comment:
I prefer to only specify the config, this hardcode the values

Reply:
Mainnet id should never change. It is supposed to be hardcoded into the code.

Reply:
> Mainnet id should never change. It is supposed to be hardcoded into the code.

But we already use config for it